src/features/features.service.spec.ts

```diff
@@ -12,16 +112,799 @@ describe("FeaturesService", () => {
+
+            const findFeatureSpy = jest
+                .spyOn(service as any, "findFeature")
+                .mockResolvedValue({
+                    ...mockFeature,
+                    addedBy: {
+                        voyageTeamId: 1,
+                    },
+                });
+            const manageOwnFeaturesByIdSpy = jest
+                .spyOn(FeaturesAbility, "manageOwnFeaturesById")
+                .mockResolvedValue();
+            prismaMock.voyageTeamMember.findFirst.mockResolvedValue(
+                mockVoyageTeamMember,
+            );
+            prismaMock.projectFeature.findMany.mockResolvedValue(
+                mockFeaturesArray,
+            );
+            prismaMock.projectFeature.update.mockResolvedValue({
+                ...mockFeaturesArray[1],
+                order: 1,
+            });
+            prismaMock.projectFeature.delete.mockResolvedValue(mockFeature);
+
+            const result = await service.deleteFeature(
+                mockFeature.id,
+                requestMock,
+            );
+
+            expect(result).toEqual({
+                message: expect.any(String),
+                status: expect.any(Number),
+            });
+            expect(findFeatureSpy).toHaveBeenCalledWith(mockFeature.id);
+            expect(prismaMock.voyageTeamMember.findFirst).toHaveBeenCalledWith({
+                where: {
+                    id: mockTeamMemberId,
+                },
+                select: {
+                    voyageTeamId: true,
+                },
+            });
+
+            expect(prismaMock.projectFeature.findMany).toHaveBeenCalledWith({
+                where: {
+                    featureCategoryId: mockFeature.featureCategoryId,
+                    addedBy: { voyageTeamId: mockTeamId },
+                },
+            });
+            expect(prismaMock.projectFeature.update).toHaveBeenCalledWith({
+                where: { id: mockFeaturesArray[1].id },
+                data: {
+                    order: mockFeaturesArray[1].order - 1,
+                },
+            });
+            expect(prismaMock.projectFeature.delete).toHaveBeenCalledWith({
+                where: {
+                    id: mockFeature.id,
+                },
+            });
+            expect(manageOwnFeaturesByIdSpy).toHaveBeenCalledWith(
+                requestMock.user,
+                mockFeature.id,
+            );
+        });
+    });
+
+    describe("updateFeatureOrderAndCategory", () => {
+        it("updateFeatureOrderAndCategory service should be defined", async () => {
+            expect(service.updateFeatureOrderAndCategory).toBeDefined();
+        });
+
+        it("should update a feature to different feature category", async () => {
+            const FeaturesArray = [
+                {
+                    id: 1,
+                    teamMemberId: 1,
+                    createdAt: new Date(Date.now()),
+                    updatedAt: new Date(Date.now()),
```

Comment:
I think we should use a constant variable that holds a specific date and time.This way we can ensure consistent results regardless of the current date, time, or timezone.
```suggestion
                    // maybe written at the top in the describe
                    const mockDate = new Date("2024-11-05T02:41:03.575Z");
                    createdAt: mockDate,
                    updatedAt: mockDate,
```